packages/location-field/src/index.tsx

```diff
@@ -0,0 +1,758 @@
+            ? "special"
+            : "normal"
+        ].push(cur);
+        return prev;
+      },
+      { special: [], normal: [] }
+    );
+
+    geocodedFeatures = [
+      ...special,
+      ...normal.sort((a, b) => {
+        if (!sortByDistance) return 0;
+        return (
+          (b.properties?.distance || Infinity) -
+          (a.properties?.distance || Infinity)
+        );
+      })
+    ];
+
+    // Add the menu sub-heading (not a selectable item)
+    // menuItems.push(<MenuItem header key='sr-header'>Search Results</MenuItem>)
+
+    // Split out different types of transit results
+    // To keep the list tidy, only include a subset of the responses for each category
+    const stopFeatures = geocodedFeatures
+      .filter(feature => feature.properties.layer === "stops")
+      .slice(0, suggestionCount);
+    const stationFeatures = geocodedFeatures
+      .filter(feature => feature.properties.layer === "stations")
+      .slice(0, suggestionCount);
+    const otherFeatures = geocodedFeatures
+      .filter(feature => feature.properties.source !== "transit")
+      .slice(0, suggestionCount);
+
+    // If no categories of features are returned, this variable is used to
+    // avoid displaying headers
+    const transitFeaturesPresent =
+      stopFeatures.length > 0 || stationFeatures.length > 0;
+
+    // Iterate through the geocoder results
+    menuItems = menuItems.concat(
+      stationFeatures.length > 0 && (
+        <S.MenuItem
+          bgColor={layerColorMap.stations}
+          centeredText
+          header
+          key="gtfs-stations-header"
+        >
+          <FormattedMessage
+            description="Text for header above Stations"
+            id="otpUi.LocationField.stations"
+          />
+        </S.MenuItem>
+      ),
+      stationFeatures.map(feature => renderFeature(itemIndex++, feature)),
+
+      stopFeatures.length > 0 && (
+        <S.MenuItem
+          bgColor={layerColorMap.stops}
+          centeredText
+          header
+          key="gtfs-stops-header"
+        >
+          <FormattedMessage
+            description="Text for header above Stops"
+            id="otpUi.LocationField.stops"
+          />
+        </S.MenuItem>
+      ),
+      stopFeatures.map(feature => renderFeature(itemIndex++, feature)),
+
+      transitFeaturesPresent && otherFeatures.length > 0 && (
+        <S.MenuItem bgColor="#333" header centeredText key="other-header">
+          <FormattedMessage
+            description="Text for header above the 'other'"
+            id="otpUi.LocationField.other"
+          />
+        </S.MenuItem>
+      ),
+      otherFeatures.map(feature => renderFeature(itemIndex++, feature))
+    );
+  }
+
+  /* 2) Process nearby transit stop options */
+  if (nearbyStops.length > 0 && !suppressNearby) {
+    // Add the menu sub-heading (not a selectable item)
+    menuItems.push(
+      <S.MenuItem header centeredText key="ns-header">
```

Comment:
Sort props (multiple instances).

Reply:
Addressed in 18d5606109e7dfce7d00d42b42c8b1b6f075a0bc